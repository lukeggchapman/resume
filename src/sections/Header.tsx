import React from 'react'
import { StyleSheet, View, Text } from '@react-pdf/renderer'
import typography from 'components/typography'
import IconLink from 'components/IconLink'

const styles = StyleSheet.create({
  header: {
    flexDirection: 'row',
    justifyContent: 'space-between',
    alignItems: 'center',
    marginBottom: 10,
  },
  info: {
    alignItems: 'flex-end',
    fontFamily: 'PT Sans',
  },
})

export default function Header() {
  return (
    <View style={styles.header}>
      <Text style={typography.h1}>Luke Chapman</Text>
      <View style={styles.info}>
        <IconLink src="mailto:lukeggchapman@gmail.com" icon="envelope" fas>
          LukeGGChapman@gmail.com
        </IconLink>
        <IconLink src="tel:+61401516361" icon="phone" fas>
          0401 516 361
        </IconLink>
      </View>
    </View>
  )
}
