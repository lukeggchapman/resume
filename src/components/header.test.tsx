import React from 'react'
import { shallow } from 'enzyme'

import Header from './header'

describe('Header', () => {
  it('renders correctly', () => {
    expect(shallow(<Header siteTitle="Test Title" />)).toMatchSnapshot()
  })
})
