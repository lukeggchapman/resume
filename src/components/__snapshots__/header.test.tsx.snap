// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`Header renders correctly 1`] = `
<header
  style={
    Object {
      "background": "rebeccapurple",
    }
  }
>
  <div
    css={
      Object {
        "map": undefined,
        "name": "1cba5ra",
        "next": undefined,
        "styles": "
        margin: 0 auto;
        max-width: 960px;
        padding: 2.25rem;
      ",
      }
    }
  >
    <mockConstructor
      css={
        Object {
          "map": undefined,
          "name": "14xgt7a",
          "next": undefined,
          "styles": "
          text-shadow: none;
        ",
        }
      }
      to="/"
    >
      <h1
        css={
          Object {
            "map": undefined,
            "name": "1fwd0ik",
            "next": undefined,
            "styles": "
            margin: 0;
            color: white;
          ",
          }
        }
      >
        Test Title
      </h1>
    </mockConstructor>
  </div>
</header>
`;
